### PR TITLE
RS/JX/Rule 5-18

### PR DESCRIPTION
**Pull request by Jiarongx-Xie**
This rule has been improved and tested with the help of Yun Joon, the ASHRAE 229 schema json file is modified to make the test work.

### CODE REVIEW

File: .idea/.gitignore
Comment (yunjoonjung-PNNL) [inline .idea/.gitignore:1]: I believe this file isn't supposed to be updated. Could you check why this file was updated?

File: .idea/inspectionProfiles/profiles_settings.xml
Comment (yunjoonjung-PNNL) [inline .idea/inspectionProfiles/profiles_settings.xml:1]: Could you also check why files under the ```inspectionProfiles``` folder was updated?

File: rct229/rulesets/ashrae9012019/section5/Pipfile
Comment (yunjoonjung-PNNL) [inline rct229/rulesets/ashrae9012019/section5/Pipfile:1]: The ```Pipfile``` and ```Pipfile.lock``` file aren't supposed to be changed. I'd appreciate it if you could check these out!

File: rct229/rulesets/ashrae9012019/section5/archive/section5rule2.py
Comment (yunjoonjung-PNNL) [inline rct229/rulesets/ashrae9012019/section5/archive/section5rule2.py:1]: Was deleting this file intended?

File: rct229/ruletest_engine/run_ruletests.py
Comment (yunjoonjung-PNNL) [inline rct229/ruletest_engine/run_ruletests.py:1]: This file shouldn't be updated! I'd appreciate it if you could reverse the change.

File: rct229/schema/ASHRAE229.schema.json [hunk 1]
```diff
@@ -559,11 +559,11 @@
                 },
                 "status_type": {
                     "description": "Choice of new, existing, addition, alteration, etc. for each ruleset.",
-                    "oneOf": [
+                    "anyOf": [
```
Comment (yunjoonjung-PNNL) [inline rct229/schema/ASHRAE229.schema.json:562]: Great job!

File: rct229/schema/ASHRAE229.schema.json [hunk 2]
```diff
@@ -559,11 +559,11 @@
...
                         {
                             "$ref": "Enumerations2019ASHRAE901.schema.json#/definitions/SpaceStatusOptions2019ASHRAE901"
                         },
-                        {
+                                                {
```
Comment (yunjoonjung-PNNL) [inline rct229/schema/ASHRAE229.schema.json:566]: These spaces also don't need to be added.

File: rct229/utils/jsonpath_utils.py [hunk 3]
```diff
@@ -11,30 +11,30 @@ def ensure_root(jpath):
     return jpath if jpath.startswith("$") else "$." + jpath
 
 
-def find_all(jpath, obj):
+def find_all(jpath, obj):  ####
```
Comment (yunjoonjung-PNNL) [inline rct229/utils/jsonpath_utils.py:14]: Please remove all the ```#```s. When we have a call, the reason why I added these was to let you know the functions we frequently used.

File: rct229/rulesets/ashrae9012019/section5/section5rule18.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            ),
+            id="5-18",
+            description="The baseline fenestration area for an existing building shall equal the existing "
+            "fenestration area prior to the proposed work.",
+            ruleset_section_title="Envelope",
+            standard_section="Section G3.1-5(c) Building Envelope Modeling Requirements for the Baseline building",
+            is_primary_rule=False,
+            each_rule=Section5Rule18.ZoneRule(),
+            index_rmr=BASELINE_0,
+            rmr_context="ruleset_model_descriptions/0",
```
Comment (yunjoonjung-PNNL) [inline rct229/rulesets/ashrae9012019/section5/section5rule18.py:32]: I believe we don't need ```rmr_context```. Could you double-check this?

File: rct229/rulesets/ashrae9012019/section5/section5rule18.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                [
+                    status_type_b in APPLICABLE_GENERAL_STATUS
+                    for status_type_b in find_all("$.spaces[*].status_type", zone_b)
+                ]
+            )
+
+        def get_calc_vals(self, context, data=None):
+            zone_b = context.BASELINE_0
+
+            has_existing_or_altered_space = any(
```
Comment (yunjoonjung-PNNL) [inline rct229/rulesets/ashrae9012019/section5/section5rule18.py:56]: I believe we can put ```_b``` at the end of this variable name.

File: rct229/rulesets/ashrae9012019/section5/section5rule18.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
+from rct229.rule_engine.partial_rule_definition import PartialRuleDefinition
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.ruleset_model_factory import produce_ruleset_model_instance
+from rct229.schema.schema_enums import SchemaEnums
+from rct229.utils.jsonpath_utils import find_all
+from rct229.rulesets.ashrae9012019 import BASELINE_0
```
Comment (yunjoonjung-PNNL) [inline rct229/rulesets/ashrae9012019/section5/section5rule18.py:6]: Could you check if you ran ```isort```?

**Review by yunjoonjung-PNNL — COMMENTED** [review]
Great 1st PR Jiarong. I left a few comments.

Some additional suggestions are

1) When you create a branch, it is recommended you name the branch name as (RS/JX/Rule5-18). This is because when you develop a bunch of rules later, and if you name the branch like ```baseline-glazing-area-check```, it'd take an additional time to find rule 5-18 branch.

2) Do you use sourcetree? This is because I saw some files were updated unexpectedly (e.g., .gitignore, etc.). This can be prevented easily using sourcetree and I believe all the team members in RCT use it.

3) Please update the RDS if there is discrepancy between the RDS and your implantation. Ideally, we'd like to have RDS and our ruleset code aligned.

Please let me know if you have any questions!

File: rct229/rulesets/ashrae9012019/section5/section5rule18.py (hunk 6 above)
Comment (Jiarongx-Xie) [reply rct229/rulesets/ashrae9012019/section5/section5rule18.py:6]: Nope. Do I need to?

File: rct229/rulesets/ashrae9012019/section5/section5rule18.py (hunk 5 above)
Comment (Jiarongx-Xie) [reply rct229/rulesets/ashrae9012019/section5/section5rule18.py:56]: Good point! Addressed:)

File: rct229/rulesets/ashrae9012019/section5/section5rule18.py (hunk 4 above)
Comment (Jiarongx-Xie) [reply rct229/rulesets/ashrae9012019/section5/section5rule18.py:32]: I deleted it and it seems to work. Good point!

File: rct229/schema/ASHRAE229.schema.json (hunk 2 above)
Comment (Jiarongx-Xie) [reply rct229/schema/ASHRAE229.schema.json:566]: Good catch!

File: rct229/rulesets/ashrae9012019/section5/archive/section5rule2.py
Comment (Jiarongx-Xie) [reply rct229/rulesets/ashrae9012019/section5/archive/section5rule2.py:1]: Weili said I should not upload any archive files or folders, so I deleted them. But when I pulled from the dev branch, these files were there. I'm not sure if we should keep it or not at this point.

File: rct229/rulesets/ashrae9012019/section5/section5rule18.py (hunk 6 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section5/section5rule18.py:6]: Yes! It needs to be done!

Comment (Jiarongx-Xie) [reply rct229/rulesets/ashrae9012019/section5/section5rule18.py:6]: Got it. Just did!

**Comment by Jiarongx-Xie** [comment]
Created another clean PR to replace this one.

**Pull request closed without merging**